Chessington.GameEngine/Pieces/Knight.cs

```diff
@@ -10,7 +10,23 @@ public Knight(Player player)
 
         public override IEnumerable<Square> GetAvailableMoves(Board board)
         {
-            return Enumerable.Empty<Square>();
+            var currentSquare = board.FindPiece(this);
+
+            var oneMinusOne = new List<int>(new[] { 1, -1 });
+            var twoMinusTwo = new List<int>(new[] { 2, -2 });
+
+            // Make set of possible jumps: (+-2, +- 1) u (+-1, +-2)
+            var colChangeJumps = oneMinusOne.SelectMany(
+                x => twoMinusTwo,
+                (rowChange, colChange) => new MoveVector(rowChange, colChange)
+            );
+            var rowChangeJumps = twoMinusTwo.SelectMany(
+                x => oneMinusOne,
+                (rowChange, colChange) => new MoveVector(rowChange, colChange)
+            );
+            var jumps = rowChangeJumps.Union(colChangeJumps);
```

Comment:
Again I would probably just write it out. Especially as I don't think you actually are saving many lines (if any) by doing it this way? And it's a fair bit more complicated 🙂

Reply:
Yep, also sorted in d4877e8